src/test/java/dal/fatih/todoly/TodolyTest.java

```diff
@@ -197,68 +200,79 @@ public void shouldNotAllowDueDateToBeEqualToNow() throws URISyntaxException {
 
     @Test
     @DirtiesContext(methodMode = DirtiesContext.MethodMode.BEFORE_METHOD)
-    public void shouldListAllTasks() throws URISyntaxException {
+    public void shouldListAllTasks() throws URISyntaxException, JsonProcessingException {
         LocalDateTime dueDate = LocalDateTime.now().plusMinutes(10);
         String title = "Listed-title-of-task", description = "Listed-description-of-task";
         DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd-MM-yyyy HH:mm:ss");
-
+        //Created three tasks
         createTask(title, description, dueDate, null, null, null);
 
         URI urlOfGetAll = new URI(createURLWithPort("/tasks"));
         ResponseEntity<String> responseEntity =
                 this.testRestTemplate.getForEntity(urlOfGetAll, String.class);
-        logger.warn(responseEntity.getBody());
 
-        String actual = responseEntity.getBody();
-        String expected = "[{\"id\":1,\"title\":\"Listed-title-of-task1\"" +
-                ",\"description\":\"Listed-description-of-task1\",\"dueDate\":\"" + dueDate.format(formatter) + "\"}" +
-                ",{\"id\":2,\"title\":\"Listed-title-of-task2\",\"description\"" +
-                ":\"Listed-description-of-task2\",\"dueDate\":\"" + dueDate.format(formatter) + "\"}" +
-                ",{\"id\":3,\"title\":\"Listed-title-of-task3\",\"description\"" +
-                ":\"Listed-description-of-task3\",\"dueDate\":\"" + dueDate.format(formatter) + "\"}]";
+        String response = responseEntity.getBody();
+
+        List<Map<String, String>> actual = jsonMapper.readValue(response, new TypeReference<List<Map<String, String>>>() {
+        });
+
+        List<Map<String, String>> expected = new ArrayList<>();
+        for (int i = 1; i < 4; i++) {
+            Map<String, String> map = new LinkedHashMap<>();
+            map.put("id", String.valueOf(i));
+            map.put("title", title + i);
+            map.put("description", description + i);
+            map.put("dueDate", dueDate.format(formatter));
+            expected.add(i - 1, map);
+        }
 
         assertThat(responseEntity.getStatusCode(), is(equalTo(HttpStatus.OK)));
-        assertThat(expected, is(equalTo(actual)));
+        assertThat(actual, is(equalTo(expected)));

```

Comment:
As I said above, Hamcrest matchers are very powerful. Use Hamcrest matchers like `hasItem`, `allOf`, `hasProperty` to facilitate your situation. An Example;

```
assertThat(
  actual, hasItem(
    allOf(
      hasProperty("id", "1"),
      hasProperty("title", title1),
      hasProperty(description, description1),
      hasProperty("dueDate", dueDate.format(formatter)),
    )
  )
);
```

You can duplicate the example I shared above for all the items in the list you want to assert on.

**Learn more about Hamcrest marchers and try to use them as much as possible**